elements/castbar.lua

```diff
@@ -314,53 +295,51 @@ local function CastStart(self, event, unit)
 	element:Show()
 end
 
-local function CastUpdate(self, event, unit, castID, spellID)
+local function CastUpdate(self, event, unit, _, _, castID)
 	local element = self.Castbar
 	if(not (element.ShouldShow or ShouldShow) (element, unit)) then
 		return
 	end
 
-	if(not element:IsShown() or element.castID ~= castID or element.spellID ~= spellID) then
+	if(not element:IsShown() or element.castID ~= castID) then
 		return
 	end
 
-	local name, startTime, endTime, _
+	local direction, duration, name, startTime, _ = Enum.StatusBarTimerDirection.ElapsedTime
 	if(event == 'UNIT_SPELLCAST_DELAYED') then
-		name, _, _, startTime, endTime = UnitCastingInfo(unit)
+		name, _, _, startTime = UnitCastingInfo(unit)
+		duration = UnitEmpoweredChannelDuration(unit)
```

Comment:
I reckon it should be `UnitCastingDuration` here.

Reply:
No this is intentional, this duration API specific to empowered casts includes the HoldAtMax endtime, something `UnitCastingDuration` doesn't do.

Reply:
Fr? Wtf? So initially we need to use `UnitChannelDuration`, but later on we have to use `UnitEmpoweredChannelDuration`? Even though it's not an empowered cast? This feels like a bug.

Reply:
Oof, you spooked me 😅